src/bootchooser.c

```diff
@@ -547,6 +614,35 @@ static gboolean uboot_set_primary(RaucSlot *slot, GError **error) {
 	return res;
 }
 
+gboolean r_boot_get_state(RaucSlot* slot, gboolean *good, GError **error)  {
+	gboolean res = FALSE;
+	GError *ierror = NULL;
+
+	g_return_val_if_fail(slot, FALSE);
+	g_return_val_if_fail(good, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	if (g_strcmp0(r_context()->config->system_bootloader, "barebox") == 0) {
+		res = barebox_get_state(slot, good, &ierror);
+	} else {
+		g_set_error(
+				error,
+				R_BOOTCHOOSER_ERROR,
+				R_BOOTCHOOSER_ERROR_NOT_SUPPORTED,
+				"Bootloader type '%s' not supported yet", r_context()->config->system_bootloader);
```

Comment:
Extend the error message to say that this only means that the state display functionality is not supported for that bootloader.